app/controllers/quizzes_controller.rb

```diff
@@ -65,8 +80,33 @@ def create
       if @quiz.save
         format.html { redirect_to quiz_url(@quiz), notice: "Quiz was successfully created." }
         format.json { render :show, status: :created, location: @quiz }
-        Student.where(teacher: current_user.email, course_id:@quiz.course_id).each do |stud|
-          Qroster.create(quiz_id:@quiz.id,student_id:stud.id)
+        if @quiz.targeted
+          recentQ = Quiz.select(:id).where(teacher:current_user.email, course_id:@quiz.course_id, completed:true).order(updated_at: :desc).limit(2).pluck(:id)
+
+          topMissed = Qroster.from(Qroster.where(quiz_id:recentQ).order(attempts: :desc)).select(:student_id).distinct.limit(10).pluck(:student_id)
```

Comment:
should we limit it to 10?

Reply:
I'm okay with anything, this was just to keep this total quiz at a max of 15 students, but we can adjust to whatever the cap is

Reply:
asking the professor would be good or we could pass it as param when creating the quiz?

Reply:
I think prof said we're good to keep normal quiz including every student, targeted quizzes taking the top 10-15 missed recently

Reply:
yeah sounds good.